src/olympia/abuse/utils.py

```diff
@@ -239,12 +262,30 @@ class CinderActionBanUser(CinderAction):
     reporter_template_path = 'abuse/emails/reporter_takedown_user.txt'
     reporter_appeal_template_path = 'abuse/emails/reporter_appeal_takedown.txt'
 
+    def should_hold_action(self):
```

Comment:
Is it not possibe to make this a method on the model of "target"? I'm probably way late to the party on this, but it feels like there's so much really complex logic in these utility classes.

Reply:
It is action dependant though - an approve decision on a UserProfile wouldn't be restricted, only a ban user.  The possible models of target are Addon, UserProfile, Rating, and Collection - I'm not really convinced it'd be an improvement to split up the logic about what is high profile among classes in 4 different files - especially in models like Addon that already have a lot of code in them.

I do agree they aren't really "utility" classes - I started a branch that renamed the file to actions, and dropped the "Cinder" from them, because they're intended to be the source of truth for review/moderation actions eventually - currently most of _that_ logic is spread among random functions in reviewers/utils.py instead.  (I never ended up merging the branch because it would always create too many conflicts with whatever patch I was working on at the time!)

Reply:
I think at a certain point it starts to be a model on it's own. But not for this PR 🤷